app/views/DirectoryView/index.js

```diff
@@ -0,0 +1,243 @@
+			const directories = await RocketChat.getDirectory({
+				query,
+				offset: data.length,
+				count: 50,
+				sort: (type === 'users') ? { username: 1 } : { usersCount: -1 }
+			});
+			if (directories.success) {
+				this.setState({
+					data: [...data, ...directories.result],
+					loading: false,
+					total: directories.total
+				});
+			} else {
+				this.setState({ loading: false });
+			}
+		} catch (error) {
+			log('err_load_directory', error);
+			this.setState({ loading: false });
+		}
+	}, 200)
+
+	search = () => {
+		this.load(true);
+	}
+
+	changeType = (type) => {
+		this.setState({ type, data: [] }, () => this.search());
+	}
+
+	toggleWorkspace = () => {
+		this.setState(({ globalUsers }) => ({ globalUsers: !globalUsers, data: [] }), () => this.search());
+	}
+
+	toggleDropdown = () => {
+		this.setState(({ showOptionsDropdown }) => ({ showOptionsDropdown: !showOptionsDropdown }));
+	}
+
+	goRoom = async({ rid, name, t }) => {
+		const { navigation } = this.props;
+		await navigation.navigate('RoomsListView');
+		navigation.navigate('RoomView', { rid, name, t });
+	}
+
+	onPressItem = async(item) => {
+		const { type } = this.state;
+		if (type === 'users') {
+			const result = await RocketChat.createDirectMessage(item.username);
+			if (result.success) {
+				this.goRoom({ rid: result.room._id, name: item.username, t: 'd' });
+			}
+		} else {
+			this.goRoom({ rid: item._id, name: item.name, t: 'c' });
+		}
+	}
+
+	renderHeader = () => {
+		const { type } = this.state;
+		return (
+			<React.Fragment>
+				<SearchBox
+					onChangeText={this.onSearchChangeText}
+					onSubmitEditing={this.search}
+					testID='federation-view-search'
+				/>
+				<Touch onPress={this.toggleDropdown} testID='federation-view-create-channel'>
+					<View style={[sharedStyles.separatorVertical, styles.toggleDropdownContainer]}>
+						<CustomIcon style={styles.toggleDropdownIcon} size={20} name={type === 'users' ? 'user' : 'hashtag'} />
+						<Text style={styles.toggleDropdownText}>{type === 'users' ? I18n.t('Users') : I18n.t('Channels')}</Text>
+						<CustomIcon name='arrow-down' size={20} style={styles.toggleDropdownArrow} />
+					</View>
+				</Touch>
+			</React.Fragment>
+		);
+	}
+
+	renderSeparator = () => <View style={[sharedStyles.separator, styles.separator]} />;
+
+	renderItem = ({ item, index }) => {
+		const { data, type } = this.state;
+		const { baseUrl, user } = this.props;
+		let style;
+		if (index === data.length - 1) {
+			style = sharedStyles.separatorBottom;
+		}
+		if (type === 'users') {
+			return (
+				<DirectoryItem
+					avatar={item.username}
+					title={item.name}
+					description={item.username}
+					rightLabel={item.federation && item.federation.peer}
+					onPress={() => this.onPressItem(item)}
+					baseUrl={baseUrl}
+					testID={`federation-view-item-${ item.name }`}
+					style={style}
+					user={user}
+					type='d'
+				/>
+			);
+		}
+		return (
+			<DirectoryItem
```

Comment:
most of the props are common in the same component above. Consider creating `commonProps` object